### PR TITLE
Implement (dump) NATS reconnection attempts on startup

### LINKED ISSUE #37: Wait for nats (and asterisk)
In k8s reconciliation model, a failing ari-proxy that triggers a container restart might indicate a problem.

Therefore, shift the reconciliation loop from the scheduler into the container run time.

~~The best place seems to be the init script.~~ Just saw: its `FROM scratch`, so a built-in reconciliation loop would probably be the right choice?

Interestingly the binary already iterates on the asterisk availability. Hare are the complete logs that triggered a restart:
```
t=2020-10-27T16:43:42+0000 lvl=info msg="starting ari-proxy server" version=5.0.1
t=2020-10-27T16:43:42+0000 lvl=eror msg="failed to connect to Asterisk" error="websocket.Dial ws://localhost:8088/ari/events?app=demo: dial tcp [::1]:8088: connect: connection refused"
t=2020-10-27T16:43:43+0000 lvl=eror msg="failed to connect to Asterisk" error="websocket.Dial ws://localhost:8088/ari/events?app=demo: dial tcp [::1]:8088: connect: connection refused"
t=2020-10-27T16:43:44+0000 lvl=eror msg="failed to connect to Asterisk" error="websocket.Dial ws://localhost:8088/ari/events?app=demo: dial tcp [::1]:8088: connect: connection refused"
t=2020-10-27T16:43:45+0000 lvl=eror msg="failed to connect to Asterisk" error="websocket.Dial ws://localhost:8088/ari/events?app=demo: dial tcp [::1]:8088: connect: connection refused"
t=2020-10-27T16:43:46+0000 lvl=eror msg="failed to connect to Asterisk" error="websocket.Dial ws://localhost:8088/ari/events?app=demo: dial tcp [::1]:8088: connect: connection refused"
Error: failed to connect to NATS: nats: no servers available for connection
Usage:
  ari-proxy [flags]
Flags:
      --ari.application string     ARI Stasis Application
      --ari.http_url string        HTTP Base URL for connecting to ARI (default "http://localhost:8088/ari")
      --ari.password string        Password for connecting to ARI
      --ari.username string        Username for connecting to ARI
      --ari.websocket_url string   Websocket URL for connecting to ARI (default "ws://localhost:8088/ari/events")
      --config string              config file (default is $HOME/.ari-proxy.yaml)
  -h, --help                       help for ari-proxy
      --nats.url string            URL for connecting to the NATS cluster (default "nats://127.0.0.1:4222")
  -v, --verbose                    Enable verbose logging
  -V, --version                    Print version information and exit
t=2020-10-27T16:43:48+0000 lvl=eror msg="server died" error="failed to connect to NATS: nats: no servers available for connection"
```

### PR DESCRIPTION
**Pull request by blaggacao**
fixes #37

This is a non-configurable ("dump") implementation. I dind't think it was worth it to expose a configuration flag for all edge cases here. If the start up fails due to non connectivity after a total of 25 seconds, then a k8s restart loop probably is within the budget.

---
This change is [Reviewable](https://reviewable.io/reviews/cycoresystems/ari-proxy/38)

### CODE REVIEW

**Comment by blaggacao** [comment]
```console
$ make ci
go mod verify
all modules verified
golangci-lint run
go build ./...
go build
go test ./...
?   	github.com/CyCoreSystems/ari-proxy/v5	[no test files]
ok  	github.com/CyCoreSystems/ari-proxy/v5/client	0.046s
ok  	github.com/CyCoreSystems/ari-proxy/v5/client/bus	0.012s
ok  	github.com/CyCoreSystems/ari-proxy/v5/client/cluster	0.056s
?   	github.com/CyCoreSystems/ari-proxy/v5/internal/integration	[no test files]
?   	github.com/CyCoreSystems/ari-proxy/v5/proxy	[no test files]
ok  	github.com/CyCoreSystems/ari-proxy/v5/server	0.058s
ok  	github.com/CyCoreSystems/ari-proxy/v5/server/dialog	0.005s
ok  	github.com/CyCoreSystems/ari-proxy/v5/session	0.002s
```

File: server/server.go [hunk 1]
```diff
@@ -73,6 +73,13 @@ func (s *Server) Listen(ctx context.Context, ariOpts *native.Options, natsURI st
 
 	// Connect to NATS
 	nc, err := nats.Connect(natsURI)
+	reconnectionAttempts := 5
+	for err == nats.ErrNoServers && reconnectionAttempts > 0 {
+		s.Log.Info("retrying to connect to NATS server", "attempts", reconnectionAttempts)
+		time.Sleep(5 * time.Second)
```
Comment (Ulexus) [inline server/server.go:79]: magic numbers:  I'd prefer that this 5s value be made into a named package-scoped constant.

File: server/server.go [hunk 2]
```diff
@@ -73,6 +73,13 @@ func (s *Server) Listen(ctx context.Context, ariOpts *native.Options, natsURI st
 
 	// Connect to NATS
 	nc, err := nats.Connect(natsURI)
+	reconnectionAttempts := 5
```
Comment (Ulexus) [inline server/server.go:76]: magic numbers:  this should be a package-scoped named constant

**Review by Ulexus — CHANGES_REQUESTED** [review]
nits and thanks!

Sorry I missed this PR.

**Comment by blaggacao** [comment]
Fixed and force-pushed, please continue review / acceptance consideration.

**Review by Ulexus — APPROVED** [review]

**Comment by Ulexus** [comment]
Thanks @blaggacao !

**Pull request merged**